ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpDataHelper.java

```diff
@@ -0,0 +1,589 @@
+                  + blobValue.getTimeToLiveInMs();
+          boolean isDeleted = blobValue.isFlagSet(IndexValue.Flags.Delete_Index);
+          numberOfKeysProcessed++;
+
+          if (blobList == null || blobList.contains(key.toString())) {
+            blobIdToMessageMap.put(key.toString(),
+                new IndexRecord(msg, isDeleted, isExpired(blobValue.getTimeToLiveInMs())));
+          }
+        }
+        if (!avoidMiscLogging) {
+          logOutput("crc " + stream.readLong());
+          logOutput("Total number of keys processed " + numberOfKeysProcessed);
+        }
+      }
+    } catch (IOException ioException) {
+      if (!avoidMiscLogging) {
+        logOutput("IOException thrown " + ioException);
+      }
+    } catch (Exception exception) {
+      if (!avoidMiscLogging) {
+        logOutput("Exception thrown " + exception);
+      }
+    }
+    return numberOfKeysProcessed;
+  }
+
+  /**
+   * Dumps all blobs in a given log file
+   * @param file the log file that needs to be parsed for
+   * @param startOffset the starting offset from which records needs to be dumped from. Can be {@code null}
+   * @param endOffset the end offset until which records need to be dumped to. Can be {@code null}
+   * @param blobs List of blobIds to be filtered for
+   * @param filter {@code true} if filtering has to be done, {@code false} otherwise
+   * @param blobIdToLogRecord {@link HashMap} of blobId to {@link LogBlobRecord} to hold the information about blobs
+   *                                         in the log after parsing
+   * @param avoidMiscLogging {@code true} if miscellaneous logging has to be avoided. {@code false} otherwise
+   * @throws IOException
+   */
+  public void dumpLog(File file, long startOffset, long endOffset, ArrayList<String> blobs, boolean filter,
+      ConcurrentHashMap<String, LogBlobRecord> blobIdToLogRecord, boolean avoidMiscLogging)
+      throws IOException {
+    logOutput("Dumping log file " + file.getAbsolutePath());
+    long currentOffset = 0;
+    RandomAccessFile randomAccessFile = new RandomAccessFile(file, "r");
+    long fileSize = file.length();
+    boolean lastBlobFailed = false;
+    if (startOffset != -1) {
+      currentOffset = startOffset;
+      randomAccessFile.seek(currentOffset);
+    }
+    if (endOffset == -1) {
+      endOffset = fileSize;
+    }
+    logOutput("Starting dumping from offset " + currentOffset);
+    while (currentOffset < endOffset) {
+      long tempCurrentOffset = currentOffset;
+      String messageheader = null;
+      String blobId = null;
+      String blobProperty = null;
+      String usermetadata = null;
+      String blobDataOutput = null;
+      String deleteMsg = null;
+      try {
+        short version = randomAccessFile.readShort();
+        if (version == 1) {
+          ByteBuffer buffer = ByteBuffer.allocate(MessageFormatRecord.MessageHeader_Format_V1.getHeaderSize());
```

Comment:
I do agree that core part where in reading from a log is same, but what we do with it differs in both. I will try to see if I can re-use them